helmfile.d/charts/crossplane/providerconfigs/templates/providerconfigs.yaml

```diff
@@ -1,15 +1,45 @@
+{{- if not .Values.clusterName }}
+{{- fail "clusterName must not be empty" }}
+{{- end }}
+---
 apiVersion: kubernetes.m.crossplane.io/v1alpha1
 kind: ClusterProviderConfig
 metadata:
-  name: mc
+  name: {{ .Values.clusterName }}
 spec:
   credentials:
     source: InjectedIdentity
 ---
 apiVersion: helm.m.crossplane.io/v1beta1
 kind: ClusterProviderConfig
 metadata:
-  name: mc
+  name: {{ .Values.clusterName }}
 spec:
   credentials:
     source: InjectedIdentity
+{{- range $cluster, $values := .Values.remoteClusters }}
+---
+apiVersion: kubernetes.m.crossplane.io/v1alpha1
+kind: ClusterProviderConfig
+metadata:
+  name: {{ $cluster }}
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      name: {{ $values.secretRef.name }}
+      namespace: {{ $values.secretRef.namespace }}
+      key: {{ $values.secretRef.key }}
+---
+apiVersion: helm.m.crossplane.io/v1beta1
+kind: ClusterProviderConfig
+metadata:
+  name: {{ $cluster }}
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      name: {{ $values.secretRef.name }}
+      namespace: {{ $values.secretRef.namespace }}
+      key: {{ $values.secretRef.key }}
+{{- end }}
```

Comment:
I started adding support for also creating the kubeconfig secrets in this Helm chart for Kubespray clusters (my hope is that it should be fine to use the CAPI kubeconfig secrets in CAPI clusters). After a while I realized that this probably should be the responsibility of Kubespray instead. Thoughts?

Reply:
I think this is something we should have a meeting about to talk about what we can do to reduce the required privileges for the providers, and maybe work more with how providers can use provided credentials.